fe/fe-core/src/main/java/com/starrocks/backup/BackupJobInfo.java

```diff
@@ -409,8 +409,13 @@ private static String[] sortTabletIds(String[] tabletIds, JSONArray tabletsOrder
             tmpList.sort((o1, o2) -> Long.valueOf(o1).compareTo(Long.valueOf(o2)));
             return tmpList.toArray(new String[0]);
         } else {
-            // the element of tabletsOrder is Long.
-            return tabletsOrder.toList().stream().map(Object::toString).toArray(String[]::new);
+            try {
+                // StarRocks uses string to deserialize tablets order json element.
+                return (String[]) tabletsOrder.toList().toArray(new String[0]);
+            } catch (ArrayStoreException e) {
+                // for compatibility with apache doris, the element of tabletsOrder is Long.
+                return tabletsOrder.toList().stream().map(Object::toString).toArray(String[]::new);
```

Comment:
this code `tabletsOrder.toList().stream().map(Object::toString).toArray(String[]::new)` can covert String and Long type.